movies/admin.py

```diff
@@ -32,20 +34,34 @@ def import_movies(self, request):
         form = CsvForm(request.POST, request.FILES)
         if form.is_valid():
             csv_file = TextIOWrapper(request.FILES['csv_file'], encoding=request.encoding)
-            movies = csv.reader(csv_file)
+            movies = csv.DictReader(csv_file)
+            added_movies = []
+            system_list = Movie.objects.values_list('pk', flat=True)
             for movie in movies:
+                created_movie = Movie(**movie)
                 try:
-                    Movie.objects.create(name=movie[0],
-                                         poster=movie[1],
-                                         description=movie[2],
-                                         imdb_id=movie[3],
-                                         length_min=int(movie[4]),
-                                         trailer_url=movie[5])
+                    name = movie["poster"].split("/")[-1]
+                    response = requests.get(created_movie.poster)
+
+                    if response.status_code == 200:
+                        created_movie.poster.save(
+                            name, ContentFile(response.content), save=False
+                        )
+                    if int(movie['pk']) not in system_list:
+                        added_movies.append(Movie(**movie))
+                    elif int(movie['pk']) in system_list:
+                        Movie.objects.bulk_update([Movie(**movie)],
+                                                  fields=['name', 'poster', 'description', 'imdb_id', 'length_min',
+                                                          'trailer_url'])
```

Comment:
The bulk_update method still seems to be called on a single object.
You can add these objects in another list (updated_movies, for example), instead.
If you are worried about the update being called on too many objects at once, after the for loop, please find a solution to call it for X objects at a time, while going through the list elements.